puppet-mode.el

```diff
@@ -632,6 +633,40 @@ of the initial include plus puppet-include-indent."
         (indent-line-to 0)))))
 
 
+;;;; SMIE
+
+(defvar puppet-use-smie nil
+  "Whether to use SMIE or not.")
+
+(defvar puppet-smie-verbose nil)
+
+(defun puppet--smie-forward-token ()
+  (smie-default-forward-token))
+
+(defun puppet--smie-backward-token ()
+  (smie-default-backward-token))
+
+(defconst puppet-smie-grammar
+  (smie-prec2->grammar
+   (smie-bnf->prec2
+    '((id)))))
+
+(defun puppet-smie-rules (kind token)
+  "Indentation rules for Puppet."
+  (when puppet-smie-verbose
+    (save-excursion
+      (let ((prev (smie-indent-backward-token))
+            (next (smie-indent-forward-token)))
+        (message "kind=%S token=%S prev=%S next=%S parent=%S"
```

Comment:
Just to be clear, you did notice it's wrapped in a conditional one needs to manually enable for anything to be printed, right? I'm beginner in elisp and not so sure how to assist with debugging properly, thought it would be helpful for the next person.